test/Interpreter/refcount_overflow.swift

```diff
@@ -0,0 +1,163 @@
+// RUN: %empty-directory(%t)
+//
+// RUN: %target-clang -x c %S/Inputs/retain_release_wrappers.c -c -o %t/retain_release_wrappers.o
+// RUN: %target-build-swift -enable-experimental-feature Extern %t/retain_release_wrappers.o %s -o %t/refcount_overflow
+// RUN: %target-codesign %t/refcount_overflow
+// RUN: %target-run %t/refcount_overflow
+
+// REQUIRES: executable_test
+// UNSUPPORTED: use_os_stdlib
+// UNSUPPORTED: back_deployment_runtime
+
+import StdlibUnittest
+
+// Declarations of runtime ABI refcounting functions.
+
+@_extern(c)
+func wrapper_swift_retain_n(_ obj: UnsafeMutableRawPointer, _ n: UInt32) -> UnsafeMutableRawPointer
+@_extern(c)
+func wrapper_swift_release_n(_ obj: UnsafeMutableRawPointer, _ n: UInt32)
+@_extern(c)
+func wrapper_swift_nonatomic_retain_n(_ obj: UnsafeMutableRawPointer, _ n: UInt32) -> UnsafeMutableRawPointer
+@_extern(c)
+func wrapper_swift_nonatomic_release_n(_ obj: UnsafeMutableRawPointer, _ n: UInt32)
+
+@_extern(c)
+func wrapper_swift_unownedRetain_n(_ obj: UnsafeMutableRawPointer, _ n: UInt32) -> UnsafeMutableRawPointer
+@_extern(c)
+func wrapper_swift_unownedRelease_n(_ obj: UnsafeMutableRawPointer, _ n: UInt32)
+@_extern(c)
+func wrapper_swift_nonatomic_unownedRetain_n(_ obj: UnsafeMutableRawPointer, _ n: UInt32) -> UnsafeMutableRawPointer
+@_extern(c)
+func wrapper_swift_nonatomic_unownedRelease_n(_ obj: UnsafeMutableRawPointer, _ n: UInt32)
+
+
+// Maximum legal retain count.
+// 32-2 bits of extra retain count, plus 1 for the implicit retain.
+let maxRC = (1 as UInt32) << (32 - 2)
+
+// Maximum retain count that fits inline on 32-bit.
+let maxInlineRC32 = (1 as UInt32) << 22
+
+// Maximum unowned count that fits inline on 32-bit.
+let maxInlineURC32 = (1 as UInt32) << 7
+
+var didDeinit = false
+
+class C {
+  deinit {
+    didDeinit = true
+  }
+}
+
+let RefcountOverflowTests = TestSuite("RefcountOverflow")
+
+RefcountOverflowTests.test("retain") {
+  do {
+    let obj = C()
+    _ = wrapper_swift_retain_n(unsafeBitCast(obj, to: UnsafeMutableRawPointer.self), maxRC - 2)
```

Comment:
I thought this was the right value, but it turns out it was just ARC inserting surplus retains when optimizations are off. I've fixed the tests to immediately convert the object to a raw pointer and do all work with that to avoid ARC causing trouble.